raiden-cli/src/routes/userDeposit.ts

```diff
@@ -0,0 +1,171 @@
+  return {
+    transaction_hash: transactionHash,
+    planned_withdraw_block_number: withdrawPlan!.block, // The plan must exists here.
+  };
+}
+
+async function withdraw(this: Cli, amount: string): Promise<WithdrawResponse> {
+  const transactionHash = await this.raiden.withdrawFromUDC(amount);
+  return { transaction_hash: transactionHash };
+}
+
+async function determineAndExecuteRequestedInteraction(
+  this: Cli,
+  request: Request,
+): Promise<DepositResponse | PlanWithdrawResponse | WithdrawResponse> {
+  if (shouldDeposit(request)) {
+    const totalDeposit = getTotalDepositAmount(request)!; // Can't be undefined here.
+    return await deposit.call(this, totalDeposit);
+  } else if (shouldPlanWithdraw(request)) {
+    const amount = getPlannedWithdrawAmount(request)!; // Can't be undefined here.
+    return await planWithdraw.call(this, amount);
+  } else if (shouldWithdraw(request)) {
+    const amount = getWithdrawAmount(request)!; // Can't be undefined here.
+    return await withdraw.call(this, amount);
+  } else {
+    throw MalformedRequestFormat;
+  }
+}
+
+async function getUDCWithdrawPlan(
+  this: Cli,
+  _request: Request,
+  response: Response,
+): Promise<void> {
+  const withdrawPlan = await this.raiden.getUDCWithdrawPlan();
+
+  if (withdrawPlan !== undefined) {
+    response.json({ ...withdrawPlan, amount: withdrawPlan.amount.toString() });
+  } else {
+    response.send();
+  }
+}
+
+async function getUDCTotalDeposit(
+  this: Cli,
+  _request: Request,
+  response: Response,
+): Promise<void> {
+  const totalDeposit = await this.raiden.getUDCTotalDeposit();
+  response.send(totalDeposit.toString());
+}
+
+async function getUDCCapacity(this: Cli, _request: Request, response: Response): Promise<void> {
+  const capacity = await this.raiden.getUDCCapacity();
+  response.send(capacity.toString());
+}
+
+async function interactWithUDC(this: Cli, request: Request, response: Response): Promise<void> {
+  try {
+    const interactionResponse = await determineAndExecuteRequestedInteraction.call(this, request);
+    response.json(interactionResponse);
+  } catch (error) {
+    if (
+      isInvalidParameterError(error) ||
+      error === MalformedNumberValue ||
+      error === MalformedRequestFormat
+    ) {
+      response.status(400).send(error.message);
+    } else if (isInsuficientFundsError(error)) {
+      response.status(402).send(error.message);
+    } else if (isConflictError(error) || error === TotalDepositTooLowError) {
+      response.status(409).send(error.message);
+    } else {
+      response.status(500).send(error.message);
+    }
+  }
+}
+
+/**
+ * @param this - Cli object
+ * @returns Router instance
+ */
+export function makeUserDepositRouter(this: Cli): Router {
+  const router = Router();
+
+  router.get('/withdraw_plan', getUDCWithdrawPlan.bind(this));
+  router.get('/total_deposit', getUDCTotalDeposit.bind(this));
+  router.get('/capacity', getUDCCapacity.bind(this));
```

Comment:
Nitpick, as a personal taste, feel free to ignore if you don´t agree: do we need 2 endpoints for those? I think `router.get('/')` returning a `{ total_deposit, balance }` object would be better.
It´s better to call it `balance` here, since this is the naming used in the webUI and PC REST API where we usually call it capacity. Also, most/all APIs return objects instead of plain numbers directly, so returning an object with the two values would be better to keep consistency. Finally, I think this is simple enough to be a single endpoint. Even though those are async requests, they´re fast as they fetch it from stored value in the state machine.

Reply:
Hmm. I thought about it but I rejected it. But if you know also have the same idea it is maybe not so bad. I just do it as I have no strong opinion here.

Reply:
✔️